src/middleware/router.js

```diff
@@ -220,15 +220,23 @@ function sendRequestToRoutes (ctx, routes, next) {
             logger.info(`executing primary route : ${route.name}`)
             if (response.headers != null && response.headers['content-type'] != null && response.headers['content-type'].indexOf('application/json+openhim') > -1) {
               // handle mediator reponse
-              const responseObj = JSON.parse(response.body)
-              ctx.mediatorResponse = responseObj
+              let payload = ''
+              response.body.on('data',function(data){
+                payload += data.toString()
+              })
+
+              response.body.on('end',function(){
```

Comment:
Also here, please use lambda instead of function.

Reply:
Addressed in 1867f6f